GOVERNANCE.rst

```diff
@@ -0,0 +1,183 @@
+* how we curate and select new contributions and existing functionality,
+* how we make decisions,
+* how we acknowledge contributions.
+
+
+Code of Conduct
+---------------
+We value the participation of every member of our community and want to ensure an that every contributor has an enjoyable and fulfilling experience. Accordingly, everyone who participates in the sktime project is expected to show respect and courtesy to other community members at all times.
+
+To make clear what is expected, we ask all members of the community to
+conform to our `Code of Conduct <https://github
+.com/alan-turing-institute/sktime/blob/master/CODE_OF_CONDUCT.rst>`_.
+
+Authorship and maintainership of algorithms
+-------------------------------------------
+
+Authors and maintainers of algorithms have special rights and duties with regard to changes and maintenance of their algorithms and associated documentation, including doc strings and tutorial notebooks.
+
+Algorithms are encapsulated in classes. To faciliate authorship and maintainership questions, each algorithm is written in a separate file.
+
+The author is the contributor who implements and contributes the algorithm to the project. The maintainer is the contributor who is responsible for maintaining an algorithm and the first point of contact for users and other contributors for all issues, questions, and proposals regarding their algorithm. The author automatically becomes the first maintainer of the algorithm.
+
+* **Veto right.** The maintainer has the right to veto any proposed changes to their algorithm within a time frame of 1 week. This does not extend to change to the common framework and API.
+
+* **Maintenance duty.** The maintainer is responsible for maintaining an algorithm. This includes understandable and instructive documentation, bugs fixes, unit testing, coding quality and style improvements, and compliance with sktime's common API. Maintainers should reply to a new issue on their algorithm within 1 week, and demonstrate a significant effort to fix any issue within 1 month.
+
+Any maintainer is expected to give up their role if they can no longer fulfil their responsibilities. If a  maintainer fails to fulfil their responsibilities in given time frame, the maintainer loses their role and all rights that come with it.
+
+Note that authors and maintainers do not *own* the algorithms or files they
+contributed or maintain. sktime is an open-source project, and all code is contributed under our 3-clause-BSD license. By contributing to sktime, ownership is transferred to all sktime developers and the project as a whole.
+
+
+Curation and inclusion criteria for algorithms
+----------------------------------------------
+
+Curation is about how we select contributions, which criteria we use in order to decide which contributions to include, and in which cases we deprecate and remove contributions.
+
+We have the following guidelines:
+
```

Comment:
my opinion differs from this:
* let´s include algorithms of whichever nature as long as they comply with the task model (i.e., don´t "cheat" by leakage etc)
* let´s have two or more tiers of algorithms: contributor archive, and a "curated collection" of good starter choices

Why: 
* the criteria below are a bit fuzzy. What does "practically useful" mean and who decides?
* for the current selection of algorithms, not all the criteria would be fulfilled! Would we have to kick them out?
What happens if a benchmarking study shows comparative performance, but that was based on a typo in the code?
*  from a game theoretic perspective, this may also discourage contributors, as it might set bars high - write a scientific paper with a full benchmarking study!
* some "classical" algorithms have not necessary been tested in the required way; also, there is scientific interest to keep some historically important methods for comparison - e.g., ARIMA.
* some methods have advantages that are not directly reflected in predictive performance, e.g., algorithms that are interpretable
* combination into a single class is not always a good thing - the extreme is a master class where everything is just determined by setting of a massive set of hyper-parameters. I think there is a good case to single out algorithms of high importance, even if they are special cases.
* transformers are not directly linked to performance! Performance is of the pipeline they are part of. How is that measured?

My preferred solution would be:
* be very open to contributions, accept everything that complies with interface and task specification
* regularly publish a curated sheet of "starter algorithms & pipelines"
* different people can publish their selection of preferred algorithm. If there´s disagreement between active sklearn contributors, there can be such seletions on equal footing (should link to reasons or studies optimally).
* perhaps multiple tiers or multiple dependent packages (e.g., contrib style)

Reply:
I'm not opposed to your model, but I'm not sure how to put it into practice it without increasing the burden on the maintainers. I'm also in favour of interpreting these criteria quite flexibly, especially at the beginning

* "practically useful" is vague, but gives us a lot of leeway in deciding which algorithms to include in the core package, e.g. it's easy to argue that classical or interpretable algorithms are practically useful, same for transformers, core developers could always vote on this
* "We strive to consolidate existing functionality when feasible" - yes, we should say that we only try to do this when it makes sense from a user and/or developer perspective, i.e. maintenance and usability considerations, rather than feasibility 
* I'm happy to accept companion package with any method/algorithm for specific methods into the sktime GitHub group, the criteria presented here are for the core repo

Reply:
>"practically useful" is vague, but gives us a lot of leeway in deciding which algorithms to include in the core package, e.g. it's easy to argue that classical or interpretable algorithms are practically useful, same for transformers, core developers could always vote on this

I still think it's a rabbit hole and it puts both too much of a burden and blame/responsibility with the TC. 
There is also a chicken-and-egg problem - if you don't have a robust implementation, you cannot easily check usefulness (e.g., via benchmarking experiment).

>I'm happy to accept companion package with any method/algorithm for specific methods into the sktime GitHub group, the criteria presented here are for the core repo

Given this, why not say that happy for companion packages in general, as long as specs (API, task compatibility, test pass) for the algorithm are met. "upgrade" into the core packages is based on some criteria which are outlined. "community interest" could also be one, e.g., for highly experimental algorithms with a large "fan base" who needs them in experiments for papers? Similar to how wikipedia or newspaper may cover content simply because it is of public interest.